onboard/Joint.h

```diff
@@ -0,0 +1,87 @@
+#ifndef JOINT_H_
```

Comment:
I'm not sure why you renamed Joint.hpp to Joint.h, the extension doesn't matter. But when you move files, please commit the rename separately from the changes you make to the file. Otherwise, in the diff it shows up as a completely new file and it's harder to follow changes.